src/Table.d.ts

```diff
@@ -2,13 +2,13 @@
     Table.d.ts -- Hand crafted type defintions for Table
 */
 
-import { AnyEntity, AnyModel, Model, OneIndexSchema, OneParams, OneProperties, OneModelSchema, OneSchema, Paged} from "./Model";
+import { AnyEntity, AnyModel, Model, OneIndexSchema, OneParams, OneProperties, OneModelSchema, OneSchema, Paged, Entity} from "./Model";
 
 export type EntityGroup = {
     [key: string]: AnyEntity[]
 };
 
-type TableConstructorParams = {
+type TableConstructorParams<Schema extends OneSchema> = {
     client?: {},                    //  Instance of DocumentClient or Dynamo.
```

Comment:
Unfortunate that this is breaking compatibility. So we can't introduce this in a 2.x.

However, our plans are for 3.0 to be based on *.ts files. So this could be a good basis for that work.
We're trying to clear the decks and get a very stable 2.x before we start the work.

Are you interested in using this PR and we could start a 3.x branch?

Reply:
Will TS users who construct their table using a bare `new Table()` work or do they have to change their code to be `new Table<XX>()`?

Reply:
The Schema type here is inferred so existing code should just work its not really meant to pass down generic types.